gui/src/mainwindow.cpp

```diff
@@ -0,0 +1,1459 @@
+        ui->bboxWestLineEdit->setEnabled(bboxOverridden);
+
+        //local image selection
+        ui->localImageFileLabel->setEnabled(true);
+        ui->localImageFileLineEdit->setEnabled(true);
+        ui->localImageFileBrowseButton->setEnabled(true);
+
+        //token
+        ui->tokenLabel->setEnabled(false);
+        ui->tokenLineEdit->setEnabled(false);
+
+        //map id
+        ui->mapIdLabel->setEnabled(false);
+        ui->mapIdLineEdit->setEnabled(false);
+
+        //zoom
+        ui->zoomLabel->setEnabled(false);
+        ui->zoomSpinBox->setEnabled(false);
+
+        //downloads
+        ui->downloadsLabel->setEnabled(false);
+        ui->downloadsSpinBox->setEnabled(false);
+
+        //user credentials
+        ui->usernameLabel->setEnabled(false);
+        ui->usernameLineEdit->setEnabled(false);
+        ui->passwordLabel->setEnabled(false);
+        ui->passwordLineEdit->setEnabled(false);
+
+        //validate the old imagepath
+        on_imagepathLineEditLostFocus();
+    }
+}
+
+void MainWindow::on_viewMetadataButton_clicked()
+{
+    QMessageBox::information(
+        this,
+        tr("Metadata"),
+        tr("Viewing Metadata is currently not supported."));
+}
+
+void MainWindow::on_nmsCheckBox_toggled(bool checked)
+{
+    if(checked) {
+        ui->nmsSpinBox->setEnabled(true);
+    }
+    else {
+        ui->nmsSpinBox->setEnabled(false);
+    }
+}
+
+void MainWindow::on_bboxOverrideCheckBox_toggled(bool checked)
+{
+    if(checked) {
+        ui->bboxNorthLineEdit->setEnabled(true);
+        ui->bboxSouthLineEdit->setEnabled(true);
+        ui->bboxEastLineEdit->setEnabled(true);
+        ui->bboxWestLineEdit->setEnabled(true);
+    }
+    else {
+        //set bbox values back to the local image's bounding box
+        if(hasValidLocalImagePath) {
+            on_imagepathLineEditLostFocus();
+        }
+
+        ui->bboxNorthLineEdit->setEnabled(false);
+        ui->bboxSouthLineEdit->setEnabled(false);
+        ui->bboxEastLineEdit->setEnabled(false);
+        ui->bboxWestLineEdit->setEnabled(false);
+    }
+}
+
+void MainWindow::on_outputLocationBrowseButton_clicked()
+{
+    QString path = QFileDialog::getExistingDirectory(this, tr("Select Output Location"));
+    if(!path.isEmpty() && !path.isNull()) {
+        ui->outputLocationLineEdit->setText(path);
+        //manually invoke the slot to check the new directory path
+        on_outputLocationLineEditLostFocus();
+    }
+}
+
+void MainWindow::on_helpPushButton_clicked()
+{
+    QMessageBox::information(
+        this,
+        tr("Help"),
+        tr("Help is currently not supported."));
+}
+
+void MainWindow::on_runPushButton_clicked()
+{
+    QString errorBuffer;
+    ui->runPushButton->setEnabled(false);
+    bool validJob = validateUI(errorBuffer);
+
+    if(!validJob) {
+        ui->runPushButton->setEnabled(true);
+        QMessageBox::critical(
+            this,
+            tr("Error"),
```

Comment:
concur.  Should be "Error", "Warning", whatever for the title The meat is the text inside the box.